createst.py

```diff
@@ -0,0 +1,400 @@
+    elif len(rules) == 1:
+        largs += ["-S", rules[0]]
+
+    return largs
+
+
+def create_env():
+    """
+    Return environment for the test being created.
+    """
+    test_output_dir = os.path.join(test_dir, "output")
+    extraenv = {
+        # The suricata source directory.
+        "SRCDIR": cwd,
+        "TZ": "UTC",
+        "output_dir": test_output_dir,
+        "ASAN_OPTIONS": "detect_leaks=0",
+    }
+    env = os.environ.copy()
+    env.update(extraenv)
+
+    return env
+
+
+## Parser
+def parse_args():
+    """
+    Parse arguments and return them to main for processing.
+    """
+    parser = argparse.ArgumentParser(
+        description="Create tests with a given PCAP. Execute the script"
+                " from a valid Suricata source directory.")
+    parser.add_argument("test-name", metavar="<test-name>",
+                        help="Name of the test folder")
+    parser.add_argument("pcap", metavar="<pcap-file>",
+                        help="Path to the PCAP file")
+    parser.add_argument("--output-path", default=DEFAULT_TEST_DIR, metavar="<output-path>",
+                        help="Path to the folder where generated test.yaml should be put")
+    parser.add_argument("--eventtype-only", default=None, action="store_true",
+                        help="Create filter blocks based on event types only")
+    parser.add_argument("--allow-events", nargs="?", default=None,
+                        help="Create filter blocks for the specified events")
+
+    # add arg to allow stdout only
+    args = parser.parse_args()
+
+    return args
+
+
+def eve2test():
+    """
+    Process the provided eve.json file and write the required checks in the
+    provided output file.
+    """
+    logger.info("Running eve2test...")
+    content = list()
+    event_types = defaultdict(int)
+    eve_path = os.path.join(test_dir, "output", "eve.json")
+    with open(eve_path, "r") as fp:
+        for line in fp:
+            eve_rule = json.loads(line)
+            content.append(eve_rule)
+            eve_type = eve_rule.get("event_type")
+            event_types[eve_type] += 1
+    if args["eventtype_only"]:
+        if args["allow_events"]:
+            logger.warning("--allow-events shall not be used with --eventtype-only")
+        filter_event_type(event_types=event_types)
+        return
+    filter_event_type_params(eve_rules=content)
+
+
+def generate_eve():
+    """
+    Create eve.json with Suricata as per the given configuration and PCAP.
+    In case of successful run, call eve2test to convert eve.json thus
+    created into test.yaml.
+    """
+    largs = create_local_args()
+    env = create_env()
+
+    p = subprocess.Popen(
+        largs, cwd=cwd, env=env,
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = p.communicate()
+    if not len(stderr):
+        eve2test()
+    else:
+        logger.error(stderr.decode("utf-8"))
+        sys.exit(1)
```

Comment:
Probably better to use the actual exit status rather than the present of output on stderr.  After communicate, you can use `p.wait()` to get the status.

Reply:
Yes, this sounds better. Adding this in the next version.